### ISSUE TITLE
Replace use of pointers in scripts by actual script objects

### ISSUE THREAD

**Issue by DemiMarie** [weechat/weechat#864]
All supported scripting languages support using proper objects instead of strings to refer to pointers.  This provides memory safety, automatic memory management, and freedom from crashes due to buggy scripts (with a few exceptions).

**Comment by flashcode** [comment]
Hi,

That means rewrite the whole scripting API (~50k lines in WeeChat itself), and all scripts.
I understand the need, but it's huge work to rewrite all scripting APIs and almost all scripts.

Current design was chosen (many years ago) for speed reasons.

If one day the API is rewritten, this would probably be done like that, with objects. So why not in far future...

**Comment by DemiMarie** [comment]
@flashcode I don't think that all scripts would need to be rewritten.  All scripting languages supported by WeeChat are dynamically typed, and the strings currently handed out are meant to be opaque.

**Comment by DemiMarie** [comment]
Here is my idea:

- For each type of object `x` for which a pointer is passed to a script, create a new type of scripting object.
- Have WeeChat check types upon entry.
